sdk/provisioning/Azure.Provisioning/src/Expressions/BicepTypeMapping.cs

```diff
@@ -121,6 +128,30 @@ IBicepValue v when (v.Kind == BicepValueKind.Unset) => BicepSyntax.Null(),
             _ => throw new InvalidOperationException($"Cannot convert {value} to a Bicep expression.")
         };
 
+        BicepExpression FromFloat(float f)
+        {
+            // see if the value is a whole number
+            var i = (int)f;
+            if (f == i)
+            {
+                return BicepSyntax.Value(i);
+            }
+            // otherwise we use the workaround from https://github.com/Azure/bicep/issues/1386#issuecomment-818077233
+            return BicepFunction.ParseJson(BicepSyntax.Value(f.ToString())).Compile();
+        }
+
+        BicepExpression FromDouble(double d)
+        {
+            // see if the value is a whole number
+            var i = (int)d;
+            if (d == i)
```

Comment:
This isn't the safest way to check for a whole number.   I'd recommend something more like:
```suggestion
            // see if the value is a whole number
            if (Math.Floor(d) == d)
```